docs/devGuide/writingComponents.md

```diff
@@ -0,0 +1,148 @@
+{% set title = "Writing Components" %}
+<span id="title" class="d-none">{{ title }}</span>
+
+<frontmatter>
+  title: "{{ title }}"
+  layout: devGuide.md
+  pageNav: default
+</frontmatter>
+
+# {{ title }}
+
+<div class="lead">
+
+This page explains how MarkBind components work, focused on implementation and testing.
+</div>
+
+MarkBind provides a number of components (e.g. expandable panels, tooltips) dynamically express content. 
+In order to serve content on the browser, MarkBind syntax is converted to valid HTML.
+
+<panel header="How are components in MarkBind syntax parsed and converted to HTML?">
+
+The main logic of the node processing flow can be found in `NodeProcessor`.
```

Comment:
Maybe can include a link to the `NodeProcessor` file? (Or alternatively can state the directory path of the file so that it is easier for the readers to find and reference the file.)

Reply:
Added both since MarkBind link formatting doesn't show up well on `inline code`!